### ISSUE TITLE
Narrowing searches does not work as intended; need parenthesis

### ISSUE THREAD

**Issue by arthurl** [djcb/mu#846]
## Current behaviour

The present behaviour is exactly as described in the manual:

> Technically, narrowing the results of query x with expression y implies doing a search (x) AND y.

However, I think this isn't what one would expect.

## Expected behaviour

IMO, the most intuitive behaviour is to search (x) AND (y).

Consider: if I narrow the results of query x by a OR b, the resultant query would be

    (x) AND a OR b

Which turns out to be

    ((x) AND a) OR b

While it should have been

    (x) AND (a OR b)

Long story short, I think we need a set of parenthesis for the y term.

Thanks for the great work!

**Comment by djcb** [comment]
Yes, that makes sense -- I've changed it.
